crates/typst-syntax/src/ast.rs

```diff
@@ -27,13 +104,27 @@ pub trait AstNode<'a>: Sized {
     }
 }
 
+/// Implements [`AstNode`] for a struct whose name matches a [`SyntaxKind`]
+/// variant.
+///
+/// The struct becomes a wrapper around a [`SyntaxNode`] pointer, and the
+/// implementation of [`AstNode::from_untyped`] checks that the pointer's kind
+/// matches when converting, returning `Some` or `None` respectively.
+///
+/// The generated struct is the basis for typed accessor methods for properties
+/// of this ast node. For example, the [`Raw`] struct has methods for accessing
+/// its content by lines, its optional language tag, and whether the raw element
+/// is inline or a block. These methods are accessible only _after_ a
+/// `SyntaxNode` is coerced to the `Raw` struct type (via `from_untyped`).
```

Comment:
Perhaps as a complement to what was being said earlier:

```suggestion
/// `SyntaxNode` is coerced to the `Raw` struct type (via `from_untyped`),
/// guaranteeing their implementations will work with the expected structure.
```